app/models/authority.rb

```diff
@@ -401,4 +402,16 @@ def validate_linked_authority
     errors.add(:base, :no_linked_authority) if person.nil? && corporate_body.nil?
     errors.add(:base, :multiple_linked_authorities) if person.present? && corporate_body.present?
   end
+
+  # rubocop:disable Style/GuardClause
+  def validate_collection_types
```

Comment:
Additional validation: root_collection must be of 'root' type and uncollected_works_collection must be of 'uncollected' type.